sql/src/main/java/com/vmware/ddlog/DDlogJooqProvider.java

```diff
@@ -111,4 +161,72 @@ private Object structToValue(final Field<?> field, final DDlogRecord record) {
                 throw new RuntimeException("Unknown datatype %s of field %s in table %s in update received");
         }
     }
+
+    private DDlogRecord toDDlogRecord(final String tableName, final String[] args) {
+        final DDlogRecord[] recordsArray = new DDlogRecord[args.length];
+        final List<Field<?>> fields = tables.get(tableName.toUpperCase());
+
+        int fieldIndex = 0;
+        for (final Field<?> field : fields) {
+            final Class<?> cls = field.getType();
+            try {
+                // Handle nullable columns here
+                if (args[fieldIndex] == null) {
+                    recordsArray[fieldIndex] = DDlogRecord.makeStruct(DDLOG_NONE_TYPE, new DDlogRecord[0]);
+                }
+                else {
+                    switch (cls.getName()) {
+                        case BOOLEAN_TYPE:
+                            recordsArray[fieldIndex] = maybeOption(field,
+                                    new DDlogRecord(Boolean.parseBoolean(args[fieldIndex])));
+                            break;
+                        case INTEGER_TYPE:
+                            recordsArray[fieldIndex] = maybeOption(field,
+                                    new DDlogRecord(Integer.parseInt(args[fieldIndex])));
+                            break;
+                        case LONG_TYPE:
+                            recordsArray[fieldIndex] = maybeOption(field,
+                                    new DDlogRecord(Long.parseLong(args[fieldIndex])));
+                            break;
+                        case STRING_TYPE:
+                            // Strings have to be escaped by single quotes
+                            if (args[fieldIndex].startsWith("'") && args[fieldIndex].endsWith("'")) {
+                                recordsArray[fieldIndex] = maybeOption(field,
+                                        new DDlogRecord(args[fieldIndex].substring(1, args[fieldIndex].length() - 1)));
+                            } else {
+                                throw new RuntimeException("Unexpected string");
```

Comment:
it's handy to output the unexpected string for debugging.
trim() is sometimes useful too.